test/problem1_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rspec'
+require_relative '../src/problem1'
+
+describe Problem1 do
+  subject { Problem1.new.even_numbers_sum(array) }
+
+  context 'when array has even numbers' do
+    let(:array) { [2,1,3,3,4] }
+
+    it { is_expected.to eq 6 }
+  end
+
+  context 'when array doesnt have even numbers' do
+    let(:array) { [7,1,3,3] }
+
+    it { is_expected.to eq 0 }
+  end
+
+  context 'when array has only 1 number' do
+    let(:array) { [2] }
+
+    it { is_expected.to eq 2 }
+  end
+
+  context 'when array is nil' do
+    let(:array) { nil }
+
+    it { is_expected.to eq 0}
```

Comment:
Does not seem fair to me. The problem states that you have an array of size N, so it won't be null. 
Also sum of nil to equal 0 seems odd.